### PR TITLE
python3Packages.palace: init at 0.2.5

### PR DESCRIPTION
**Pull request by McSinyx**
###### Motivation for this change

Palace is used in a game of mine.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131973` at bd974acd run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.palace</li>
    <li>python39Packages.palace</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/palace/default.nix:29:0:
```
   |
29 |     description = "Pythonic Audio Library and Codecs Environment";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/palace/default.nix:17:15:
```
   |
17 |   patches = [ ./alure2-config.diff ];
   |               ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/palace/alure2-config.diff [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+index 4c9aedb..f3ed424 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -6,5 +6,5 @@ get_target_property(include_dirs Alure2::alure2 INTERFACE_INCLUDE_DIRECTORIES)
+ message("alure2_include_dirs=${include_dirs}")
+ 
+ get_target_property(link_libraries Alure2::alure2 INTERFACE_LINK_LIBRARIES)
+-get_target_property(libalure2 Alure2::alure2 IMPORTED_LOCATION_NOCONFIG)
++get_target_property(libalure2 Alure2::alure2 IMPORTED_LOCATION_RELEASE)
+ message("alure2_extra_objects=${link_libraries};${libalure2}")
```
Comment (risicle) [inline pkgs/development/python-modules/palace/alure2-config.diff:11]: Could this not have been done with `substituteInPlace`?

File: pkgs/development/python-modules/palace/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  patches = [ ./alure2-config.diff ];
+  dontUseCmakeConfigure = true;
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ cython ];
+  propagatedBuildInputs = [ alure2 ] ++ lib.optionals (pythonOlder "3.8") [
+    typing-extensions
+  ];
+
+  doCheck = false; # FIXME: tests need an audio device
```
Comment (risicle) [inline pkgs/development/python-modules/palace/default.nix:27]: 1. Do _all_ tests need an audio device or is there a sensible subset you could enable?
2. If we're not having any tests enabled, could we at least have `pythonImportsCheck`?

Comment (McSinyx) [reply pkgs/development/python-modules/palace/default.nix:27]: All tests fail without an audio device, but I added `pythonImportsCheck`.

File: pkgs/development/python-modules/palace/alure2-config.diff (hunk 1 above)
Comment (McSinyx) [reply pkgs/development/python-modules/palace/alure2-config.diff]: Thanks for the suggestion.

File: pkgs/development/python-modules/palace/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      --replace IMPORTED_LOCATION_NOCONFIG IMPORTED_LOCATION_RELEASE
+  '';
+
+  dontUseCmakeConfigure = true;
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ cython ];
+  propagatedBuildInputs = [ alure2 ] ++ lib.optionals (pythonOlder "3.8") [
+    typing-extensions
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/palace/default.nix:29]:
```suggestion
  propagatedBuildInputs = [ alure2 ]
    ++ lib.optionals (pythonOlder "3.8") [
    typing-extensions
  ];
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 131973` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.palace</li>
<li>python39Packages.palace</li>
</ul>
</details>

**Pull request merged**